src/VoltageCalibration.cc

```diff
@@ -1008,19 +1008,19 @@ void mattak::VoltageCalibration::readFitCoeffsFromFile(const char * inFile)
     {
       // Interpolating the average residuals
       int npoints_residGraph = graph_residAve[j]->GetN();
-      resid_volt[j].resize(npoints_residGraph*2-1);
-      resid_adc[j].resize(npoints_residGraph*2-1);
+      const double dV = graph_residAve[j]->GetPointX(1) - graph_residAve[j]->GetPointX(0);
+      resid_volt[j].resize(npoints_residGraph * 2 - 1);
+      resid_adc[j].resize(npoints_residGraph * 2 - 1);
       nResidPoints[j] = resid_volt[j].size();
 
       for (int i = 0; i < npoints_residGraph; i++)
       {
         resid_volt[j][i*2] = graph_residAve[j]->GetPointX(i);
         resid_adc[j][i*2] = graph_residAve[j]->GetPointY(i);
-      }
-      for (int i = 0; i < npoints_residGraph-1; i++)
-      {
-        resid_volt[j][i*2+1] = (resid_volt[j][i*2] + resid_volt[j][i*2+2])/2;
-        resid_adc[j][i*2+1] = resid_adc[j][i*2] + (resid_volt[j][i*2+1] - resid_volt[j][i*2])*(resid_adc[j][i*2+2] - resid_adc[j][i*2])/(resid_volt[j][i*2+2] - resid_volt[j][i*2]);
+
+        // Upsampling by a factor of 2
+        resid_volt[j][i*2+1] = resid_volt[j][i*2] + dV / 2;
+        resid_adc[j][i*2+1] = graph_residAve[j]->Eval(resid_volt[j][i*2+1]);
```

Comment:
This is accidentally quadratic, due to the linear search in Eval. 

It can be made better by setting kIsSortedX bit  (something like `graph_residAve[j]->SetBit(TGraph::kIsSortedX)` first so it does a binary search instead, then it won't be quadratic anymore. It would still a bit slower in principle because Eval has to search the x axis for the right index, which we in principle already know (since we're looping over X values). Since this only effects reading in coefficients from file though, the extra simplicity may be worth it

Reply:
Accidentally? Yes because it is only during read in I thought performance is not critical. But of course it is also a bit awkward to go for the slower option. ...

Reply:
yeah, it's probably not critical unless someone is doing something where they're loading a different calibration file for every event. But already our serialization strategy is quite slow. Either way setting the kIsSortedX bit is free and worth doing...

Reply:
done